gps_sensor_broadcaster/src/gps_sensor_broadcaster.cpp

```diff
@@ -0,0 +1,168 @@
+// Copyright 2025 ros2_control development team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * Authors: Wiktor Bajor, Jakub Delicat
+ */
+
+#include "gps_sensor_broadcaster/gps_sensor_broadcaster.hpp"
+#include <map>
+
+namespace
+{
+template <class... Ts>
+struct Visitor : Ts...
+{
+  using Ts::operator()...;
+};
+template <class... Ts>
+Visitor(Ts...) -> Visitor<Ts...>;
+constexpr std::size_t COVARIANCE_3x3_SIZE = 9;
+constexpr uint16_t COVARIANCE_TYPE_KNOWN = 3;
+constexpr uint16_t COVARIANCE_TYPE_DIAGONAL_KNOWN = 2;
+}  // namespace
+
+namespace gps_sensor_broadcaster
+{
+
+callback_return_type GPSSensorBroadcaster::on_init()
+try
+{
+  param_listener_ = std::make_shared<gps_sensor_broadcaster::ParamListener>(get_node());
+  params_ = param_listener_->get_params();
+  return CallbackReturn::SUCCESS;
+}
+catch (const std::exception & e)
+{
+  fprintf(stderr, "Exception thrown during init stage with message: %s \n", e.what());
+  return CallbackReturn::ERROR;
+}
+
+callback_return_type GPSSensorBroadcaster::on_configure(const rclcpp_lifecycle::State &)
+{
+  param_listener_->refresh_dynamic_parameters();
+  params_ = param_listener_->get_params();
+
+  gps_sensor_ =
+    params_.read_covariance_from_interface
+      ? GPSSensorVariant{semantic_components::GPSSensor<GPSSensorOption::WithCovariance>(
+          params_.sensor_name)}
+      : GPSSensorVariant{
+          semantic_components::GPSSensor<GPSSensorOption::WithoutCovariance>(params_.sensor_name)};
+  std::visit(
+    Visitor{
+      [this](auto & sensor) { state_names = sensor.get_state_interface_names(); },
+      [](std::monostate &) {}},
+    gps_sensor_);
+
+  return setup_publisher();
+}
+
+callback_return_type GPSSensorBroadcaster::setup_publisher()
+try
+{
+  sensor_state_publisher_ = get_node()->create_publisher<sensor_msgs::msg::NavSatFix>(
+    "~/gps/fix", rclcpp::SystemDefaultsQoS());
+  realtime_publisher_ = std::make_unique<StatePublisher>(sensor_state_publisher_);
+  realtime_publisher_->lock();
+  realtime_publisher_->msg_.header.frame_id = params_.frame_id;
+  setup_covariance();
+  realtime_publisher_->unlock();
+
+  return callback_return_type::SUCCESS;
+}
+catch (const std::exception & e)
+{
+  fprintf(
+    stderr, "Exception thrown during publisher creation at configure stage with message: %s \n",
+    e.what());
+  return callback_return_type::ERROR;
+}
```

Comment:
```suggestion
callback_return_type GPSSensorBroadcaster::setup_publisher()
{
try
{
  sensor_state_publisher_ = get_node()->create_publisher<sensor_msgs::msg::NavSatFix>(
    "~/gps/fix", rclcpp::SystemDefaultsQoS());
  realtime_publisher_ = std::make_unique<StatePublisher>(sensor_state_publisher_);
  realtime_publisher_->lock();
  realtime_publisher_->msg_.header.frame_id = params_.frame_id;
  setup_covariance();
  realtime_publisher_->unlock();

  return callback_return_type::SUCCESS;
}
catch (const std::exception & e)
{
  fprintf(
    stderr, "Exception thrown during publisher creation at configure stage with message: %s \n",
    e.what());
  return callback_return_type::ERROR;
}
}
```

Missing enclosing parenthesis